test/mocks/network/msw/create-api-interceptor.ts

```diff
@@ -8,11 +8,12 @@ import { rest } from 'msw';
 
 import { getSetupServer } from '../../../jest-setup';
 
-export const createAPIInterceptor = <T>(
+export const createAPIInterceptor = <RequestParamsType, ResponseType = never>(
 	apiAction: string,
-	extraParamProperty?: string
-): Promise<T> =>
-	new Promise<T>((resolve, reject) => {
+	extraParamProperty?: string,
+	response?: ResponseType
+): Promise<RequestParamsType> =>
```

Comment:
at the cost of introducing a breaking change, here I would rather pass an object to improve readability and usability of the function